zerver/lib/onboarding.py

```diff
@@ -187,7 +187,23 @@ def send_welcome_bot_response(send_request: SendMessageRequest) -> None:
     to welcome-bot, trigger the welcome-bot reply."""
     welcome_bot = get_system_bot(settings.WELCOME_BOT, send_request.realm.id)
     human_response_lower = send_request.message.content.lower()
+    human_user = send_request.message.sender
     content = select_welcome_bot_response(human_response_lower)
+    realm_id = send_request.realm.id
+    commands = bot_commands()
+    if (
+        commands in content
+        and Message.objects.filter(
+            realm_id=realm_id,
+            sender_id=welcome_bot.id,
+            recipient=human_user.recipient,
```

Comment:
Use `recipient_id=human_user.recipient_id,` here.

It'll save a db query to get the recipient row. We only need ID of the recipient here.

Reply:
oh yeah right